src/bgw/job_stat.c

```diff
@@ -660,18 +670,28 @@ ts_bgw_job_stat_mark_end(BgwJob *job, JobResult result)
 								  &res,
 								  ShareRowExclusiveLock))
 		elog(ERROR, "unable to find job statistics for job %d", job->fd.id);
+
+	/* Failures are always logged */
+	if (ts_guc_enable_job_execution_logging || result != JOB_SUCCESS)
```

Comment:
Nit: Since this line is repeated multiple times, it could make sense to move this kind of logic into `ts_bgw_job_stat_history_mark_end` and return if the needed conditions are not fulfilled.

Reply:
Thanks... reorganized the code in that way.